### DISCUSSION TITLE
YAML files creation and usage in ComfyUI

### DISCUSSION THREAD

**Discussion by JHNewton** [adieyal/comfyui-dynamicprompts#22] [Q&A]
I've been away from pertty much all forms of programming for close to 20 years now.  Python is a totally new creature to me, yes pun intended, and I'd never heard of YAML.  

Anyway, I've been playing with ComfyUI for a couple of months now and I really enjoy it.  I had experimented with using test file wildcards and got good results.  Then one day while stumbling around CivitAI I came across yaml wildcard files.  Now these look like a pretty powerful tool.  Of the 3 that I down loaded I could only get one to work.  So far in my endeavors, I've figured out that you can't use tabs, everything has to be done with white space. You can have hierarchies.  All of this looks to be a good thing for creating powerful wildcard files.

What I have as yet to grasp is how to actually call these wildfiles and the different hierarchies within the yaml file.  I'm not sure but it looks like the actually wildcard info is not called by the wildcard file name, like what you do with the text version of wildcard files (creatures.txt, __creatures__).  So I'm not at all sure in how to call the various levels within the yaml wildcard file.  BTW the file I downloaded from CivitAI that worked was "characters.yaml".  The other was "Billions_of_characters.yaml". - I think I got those name correct.

 Anyway, if you can sort of steer me towards some documents that make use of the yaml files. particularly as they relate to use within ComfyUI I'd greatly appreciate it.  The dynamicpromopts nodes appear to have a great deal of potential and I look forward to seeing them grow.

**Comment by adieyal** [comment]
It's been a while since you posted this. I hope you've figured it out in the meanwhile. In case not, have a look at this file: [publicprompts.yaml](https://github.com/adieyal/sd-dynamic-prompts/blob/main/collections/publicprompts.yaml) - it has comments with examples. Also, you might be interested in checking out the auto1111 version of dynamicprompts - [sd-dynamic-prompts](https://github.com/adieyal/sd-dynamic-prompts/) - it has a built-in wildcards browser which creates the wildcard string for you.
